src/frontend/data/stores/useChatItemsStore/index.ts

```diff
@@ -0,0 +1,148 @@
+import { DateTime } from 'luxon';
+import create from 'zustand';
+
+import { TIME_TRIGGER_FOR_GROUPING_MESSAGES_IN_MS } from 'default/chat';
+
+export type ReceivedMessageType = {
+  content: string;
+  sender: string;
+  sentAt: DateTime;
+};
+
+export type ChatMessageType = {
+  content: string;
+  sentAt: DateTime;
+};
+
+export type ChatMessageGroupType = {
+  messages: ChatMessageType[];
+  sender: string;
+};
+
+export enum presenceType {
+  ARRIVAL = 'arrival',
+  DEPARTURE = 'departure',
+}
+export type ChatPresenceType = {
+  receivedAt: DateTime;
+  sender: string;
+  type: presenceType;
+};
+
+export enum chatItemType {
+  GROUP_MESSAGE = 0,
+  PRESENCE = 1,
+}
```

Comment:
you should use string enum
You can face issue with number enum (or enum without value since integer value will be assign by default).
For instance:
```
interface T {
  value?: chatItemType;
}

function test(a: T) {
  if (a.value) {
    //  do some stuff
  }
}
```
here your code will compile and raise no warning, but objects with `value` assigned to `chatItemType.GROUP_MESSAGE` will not enter the if statement

Reply:
Ok thanks for the tip !